src/CSnakes.Runtime/Python/GeneratorIterator.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+
+namespace CSnakes.Runtime.Python;
+public class GeneratorIterator<TYield, TSend, TReturn> : IGeneratorIterator<TYield, TSend, TReturn>
+{
+    private readonly PyObject generator;
+    private readonly PyObject nextPyFunction;
+    private readonly PyObject closePyFunction;
+    private readonly PyObject sendPyFunction;
+
+    private TYield current = default!;
+
+    public GeneratorIterator(PyObject generator)
+    {
+        this.generator = generator;
+        nextPyFunction = generator.GetAttr("__next__");
+        closePyFunction = generator.GetAttr("close");
+        sendPyFunction = generator.GetAttr("send");
+    }
+
+    public TYield Current => current;
+
+    object IEnumerator.Current => Current!;
+
+    public void Dispose()
+    {
+        generator.Dispose();
+        nextPyFunction.Dispose();
+        using PyObject close = closePyFunction.Call();
+        closePyFunction.Dispose();
+        sendPyFunction.Dispose();

```

Comment:
Is the order here important? If not, I would write `closePyFunction.Call().Dispose()`. If it is, perhaps a comment on why?